test/plugins/test_hook.py

```diff
@@ -13,162 +13,98 @@
 # included in all copies or substantial portions of the Software.
 
 
+from __future__ import annotations
+
 import os.path
 import sys
-import tempfile
 import unittest
+from contextlib import contextmanager
+from typing import Callable, Iterator
 
-from beets import config, plugins
-from beets.test import _common
-from beets.test.helper import TestHelper, capture_log
-
-
-def get_temporary_path():
-    temporary_directory = tempfile._get_default_tempdir()
-    temporary_name = next(tempfile._get_candidate_names())
-
-    return os.path.join(temporary_directory, temporary_name)
+from beets import plugins
+from beets.test.helper import PluginTestCase, capture_log
 
 
-class HookTest(_common.TestCase, TestHelper):
-    TEST_HOOK_COUNT = 5
-
-    def setUp(self):
-        self.setup_beets()
-
-    def tearDown(self):
-        self.unload_plugins()
-        self.teardown_beets()
+class HookTestCase(PluginTestCase):
+    plugin = "hook"
+    preload_plugin = False
 
-    def _add_hook(self, event, command):
-        hook = {"event": event, "command": command}
+    def _get_hook(self, event: str, command: str) -> dict[str, str]:
+        return {"event": event, "command": command}
 
-        hooks = config["hook"]["hooks"].get(list) if "hook" in config else []
-        hooks.append(hook)
 
-        config["hook"]["hooks"] = hooks
+class HookLogsTest(HookTestCase):
+    @contextmanager
+    def _configure_logs(self, command: str) -> Iterator[list[str]]:
+        config = {"hooks": [self._get_hook("test_event", command)]}
 
-    def test_hook_empty_command(self):
-        self._add_hook("test_event", "")
-
-        self.load_plugins("hook")
-
-        with capture_log("beets.hook") as logs:
+        with self.configure_plugin(config), capture_log("beets.hook") as logs:
             plugins.send("test_event")
+            yield logs
 
-        self.assertIn('hook: invalid command ""', logs)
+    def test_hook_empty_command(self):
+        with self._configure_logs("") as logs:
+            self.assertIn('hook: invalid command ""', logs)
 
     # FIXME: fails on windows
     @unittest.skipIf(sys.platform == "win32", "win32")
     def test_hook_non_zero_exit(self):
-        self._add_hook("test_event", 'sh -c "exit 1"')
-
-        self.load_plugins("hook")
-
-        with capture_log("beets.hook") as logs:
-            plugins.send("test_event")
-
-        self.assertIn("hook: hook for test_event exited with status 1", logs)
+        with self._configure_logs('sh -c "exit 1"') as logs:
+            self.assertIn(
+                "hook: hook for test_event exited with status 1", logs
+            )
 
     def test_hook_non_existent_command(self):
-        self._add_hook("test_event", "non-existent-command")
+        with self._configure_logs("non-existent-command") as logs:
+            self.assertTrue(
+                any(
+                    message.startswith("hook: hook for test_event failed: ")
```

Comment:
Can we implement this as `self.assertIn` instead?  What does the error message contained here look like, and how does it vary?  If this style is necessary, a comment explaining the variation in the message is needed.

Reply:
I think this kind of improvement may be out of scope of the structural change I did here

Reply:
Alright, could you leave a TODO comment so that somebody can pick it up in the future?

Reply:
Had a quick look into this and got it done!